airbyte-integrations/connectors/destination-shelby/src/main/kotlin/http/job/JobRepository.kt

```diff
@@ -0,0 +1,129 @@
+        }
+
+        val salesforceJobStatus: String = jsonDecoder.decode(response).get("state").asText()
+        if (salesforceJobStatus in TERMINAL_STATES) {
+            if (salesforceJobStatus in FAILED_STATES) {
+                job.status = JobStatus.INCOMPLETE
+            } else {
+                job.status = JobStatus.COMPLETE
+            }
+        }
+    }
+
+    fun printFailedRecords(job: Job) {
+        val response: Response = httpClient.sendRequest(
+            Request(
+                method = RequestMethod.GET,
+                url = "$baseUrl/services/data/v62.0/jobs/ingest/${job.id}/failedResults",
+            )
+        )
+        val decodedResponse = csvDecoder.decode(response)
+        // FIXME we need to redact the record content for now if we decide to print it
+        decodedResponse.forEach { print(it) }
+    }
+
+    private fun createSalesforceJob(stream: DestinationStream): String {
+        val bodyNode: ObjectNode = Jsons.objectNode()
+            .put("object", stream.destinationObjectName)
+            .put("operation", assembleOperation(stream.importType))
+        if (stream.importType is Update) {
+            // for the externalIdFieldName, we get [0] here because we know the matchingKeys can't be composite for Salesforce
+            bodyNode.put("externalIdFieldName", (stream.importType as Update).matchingKey[0])
+        }
+        val response: Response = httpClient.sendRequest(
+            Request(
+                method = RequestMethod.POST,
+                url = "$baseUrl/services/data/v62.0/jobs/ingest",
+                headers = mapOf("Content-Type" to "application/json"),
+                body = bodyNode.toString().toByteArray(Charsets.UTF_8)
+            )
+        )
+        if (response.statusCode != 200) {
+            throw IllegalStateException("Invalid response with status code ${response.statusCode} while creating job: ${jsonDecoder.decode(response).toPrettyString()}")
+        }
+
+        return jsonDecoder.decode(response).get("id").asText()
+    }
+
+    private fun uploadBatch(jobId: String, batch: Batch) {
+        val response: Response = httpClient.sendRequest(
+            Request(
+                method = RequestMethod.PUT,
+                url = "$baseUrl/services/data/v62.0/jobs/ingest/$jobId/batches",
+                headers = mapOf("Content-Type" to "text/csv"),
+                body = batch.toRequestBody()
+            )
+        )
+
+        if (response.statusCode != 201) {
+            throw IllegalStateException("Invalid response with status code ${response.statusCode} while uploading batch: ${jsonDecoder.decode(response).toPrettyString()}")
+        }
+    }
+
+    private fun assembleOperation(importType: ImportType): String {
+        return when (importType) {
+            is Append -> "insert"
```

Comment:
Not sure how much we want to reference those strings, but I'd lean towards string constant to ease further usage tracking of the different values.